pkg/client/client.go

```diff
@@ -0,0 +1,72 @@
+package client
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+)
+
+// ObjectKey identifies a Kubernetes Object.
+type ObjectKey types.NamespacedName
```

Comment:
I am still thinking if it is an overkill ? because passing in `namespace` and `name` as is is less verbose.

Reply:
Yeah, I've had a very long discussion with Phil about that, still believe that `types.NamespacedName` is more explicit than `ObjectKey`.

Reply:
what's wrong with taking `namespace, name string`?

Reply:
Nothing wrong with taking `namespace` and `name` input, but I want to look at it from `ctrl` pkg's point of view as well because this pkg is going to be consumed by the `ctrl` pkg and would like it to be consistent.

Reply:
Plus, taking `NamespacedName` makes it a *bit* harder to screw up the ordering of the two, especially if you have a linter that warns about initializing structs without field names.

Reply:
perhaps it should be an alias though `type ObjectKey = types.NamespacedName`